packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/GoogleMapMarkerController.m

```diff
@@ -83,15 +89,31 @@ - (void)setRotation:(CLLocationDegrees)rotation {
 }
 
 - (void)setVisible:(BOOL)visible {
-  self.marker.map = visible ? self.mapView : nil;
+  // If marker belongs the cluster manager, visibility need to be controlled with the opacity
+  // as the cluster manager controls when marker is on the map and when not.
+  // Alpha value for marker must always be interpreted before visibility value.
+  if (self.clusterManagerId && self.clusterManagerId != (id)[NSNull null]) {
```

Comment:
Similar casting comment here: if code is passing `NSNull` to the initializer, which says it takes an `NSString*`, then the calling code is wrong and should be fixed.